File: first.java
import java.io.File;
import java.io.FileNotFoundException;
import java.util.Scanner;
// FUNCTION TO CHECK IF INTEGER VALUE IS PRIME
public class first{
    public static boolean isPrime(int val){
        if(val == 3){
            return true;
        }else{
            for(int i = 2; i<val/2; i++){
                if(val%i == 0){
                    return false;
                }
            }
        }
        return true;
    }
    // OPEN AND READ INTS FROM FILE TO FIND IF VALUE IS PRIME AS WELL AS IF VALUE-2 IS PRIME OR VALUE+2 IS PRIME
    public static void main(String[] args) {
        try{
            File read = new File("first.txt");
            Scanner input = new Scanner(read);
            while(input.hasNext()){
                String[] list = input.nextLine().split(" ");
                int temp = Integer.parseInt(list[0]);
                if(isPrime(temp) == false){
                    System.out.println("no");
                }else if(isPrime(temp-2) == true || isPrime(temp+2) == true){
                    System.out.println("yes");
                }else{
                    System.out.println("no");
                }
            }
            input.close();
        } catch (FileNotFoundException f){
            System.out.println("Error, the file is not found");
            f.printStackTrace();
        }
        
    }
}

File: first.c
#include <stdio.h>
#include <stdlib.h>
// CHECK TO SEE IF A VALUE IS PRIME
int isPrime(int value){
    if(value<=2){
        return 0;
    }else{
        for(int i = 2; i<(value/2); i++){
            if(value%i == 0){
                return 0;
            }
        }
        return 1;
    }
}
// READ INTS FROM A FILE AND FIND COUNT OF THE TIMES THE VALUE IS PRIME AS WELL AS IF VALUE-2 IS PRIME OR VALUE+2 IS PRIME
int main(int argc, char *filename[argc+1]){
    FILE *file = fopen(filename[1], "r");
    if(file == NULL){
        EXIT_FAILURE;
    }
    int value;
    int counter = 0;
    while(fscanf(file, "%d", &value) != EOF){
        int prime = isPrime(value);
        counter++;
        if(prime == 1){
            if(isPrime(value+2) == 1 || isPrime(value-2) == 1){
                printf("yes\n");
            }else{
                printf("no\n");
            }
        }else{
            printf("no\n");
        }
    }
    fclose(file);
    return EXIT_SUCCESS;
}
